gateway-js/src/datasources/RemoteGraphQLDataSource.ts

```diff
@@ -192,7 +192,8 @@ export class RemoteGraphQLDataSource<TContext extends Record<string, any> = Reco
   public didEncounterError(
     error: Error,
     _fetchRequest: Request,
-    _fetchResponse?: Response
+    _fetchResponse?: Response,
+    _context?: TContext,
```

Comment:
It's a shame that this function doesn't take options (named) arguments but given that it doesn't, this seems like the best change.

Reply:
Agree - I think it would be nice to move to an options object in a breaking change in the future. Should I open an issue?

Reply:
🤷 there are deeper issues with this API (like how the class name sounds like the not really related DataSource and how the method names sound like the AS plugin API) that might be worth addressing at some point